client/src/components/_common/Button/Button.jsx

```diff
@@ -39,6 +39,33 @@ const Button = ({
     }
   }
 
+  // Manage prop warnings
+  /* eslint-disable no-console */
+  if (type === 'link' && size) {
+    size = '';
+    // Component will work, except `size` is ineffectual
+    console.warn('A <Button type="link"> ignores `size` prop.');
+  }
+  if (type === 'primary' && size === 'small') {
+    type = 'secondary';
+    // Component will work, except `type` is overridden
+    console.error(
+      'A <Button type="primary" size="small"> is not allowed. ' +
+        'Using `type="secondary"` instead.'
+    );
+  }
+  if (type !== 'link' && !size) {
+    size = 'short';
+    // Component will work, except `size` is auto-set
+    console.debug(
+      'A <Button> that is not `type="link"` and has no `size` ' +
+        'is automatically assigned `size="short"`.'
+    );
+  }
```

Comment:
@wesleyboar Could you add some tests to cover these expectations?

Reply:
Ah, good call. Yes. I will.

Reply:
Don't worry about testing the console output, just the size and type based on input params

Reply:
Coming in new PR, for diff legibility and PR distinction.

Reply:
New (sub) PR for test cases: https://github.com/TACC/Core-Portal/pull/640

Reply:
Ooh, after merge of this PR, the base for #640 (this FP-1491 branch) was auto-updated to `main`. GitHub do good.